### ISSUE TITLE
Implement/test triangle and gwesp terms with networkLite API

### ISSUE THREAD

**Issue by smjenness** [statnet/tergmLite#28]
As discussed on our call yesterday, @chad-klumb could you please implement the two ergm terms triangle and gwesp on the allowed terms list. This should be implemented on the current master branch before you branch off a new dev branch. This will round out the set of standard terms we use in both research and workshop teaching. Thanks!

cc: @martinamorris

**Comment by chad-klumb** [comment]
The `InitErgmTerm` functions for `triangle` and `gwesp` appear to have no problem working with `networkLite`s.

Regarding `gwesp` though, there may be larger issues about curved terms; sometimes those need to be handled differently than non-curved terms.  I'm not sure if all that has been built into `EpiModel`/`tergmLite`'s typical workflow or not.

Do you want to use `gwesp` as a curved term (with `fixed=FALSE`) or only as a non-curved term (`fixed=TRUE`, which I believe is typically what we do in workshops, as `fixed=FALSE` can be difficult to fit)?

**Comment by smjenness** [comment]
@martinamorris @sgoodreau please weigh in here

**Comment by martinamorris** [comment]
First -- we'll only use this for courses, as it requires complete network data.  Pavel/Michal/I are working on an ergm.ego paper for triadic estimation when the alter-alter matrix data are available, but it's proving to be slow going.

Second -- it's almost impossible to estimate the decay parameter with a single cross-sectional network.  In the recent Stewart/Schweinberger/Morris paper we revewed the lit and only found one example (the Lazega data) where the decay parameter was fit rather than fixed.  Stewart/Schweinberger have another paper showing why, but the bottom line is that there's typically not enough info in one net, so decay estimation generally requires multi-level data (replicates).

So ... I think it'd be fine to make the gwesp(fixed) available. 
 * add a note in the terms manual about the restriction to fixed when using networklite objects
 * if possible to have an informative error if someone uses fixed=F in that context, that would be great.

**Comment by martinamorris** [comment]
@chad-klumb would you please evaluate/report back on all of the triad terms?  i think this will get them:

> search.ergmTerms(categories = "triad-related")

**Comment by chad-klumb** [comment]
`localtriangle` takes an edge attribute, and `transitiveweights` is weighted only, so I would not expect either of those to work

the remainder should in principle work with the addition of one or two trivial `networkLite` accessors (like `network.edgecount.networkLite`)

however many of the triad-related terms are directed only; from the front page of `tergmLite`:

For epidemic models, the only types of networks that we typically estimate and simulate from are undirected, binary edge networks with no missing data (as it is simulated).

This makes me think that portions of `tergmLite` and/or `EpiModel` may (possibly) not work with directed networks. @sgoodreau @smjenness do you know if anything in either package (or related packages) is plausibly specific to undirected networks?

To be clear, getting through `InitErgmTerm` is only one piece of the puzzle; I can test `networkLite`s and `network`s for equivalent `InitErgmTerm` behavior, but that alone isn't a guarantee that everything else will handle things appropriately.

**Comment by smjenness** [comment]
@chad-klumb : you're right, epidemic simulations of the type we run are only ever on undirected binary networks, and so tergmLite and EpiModel were designed only around that network type. I see no need in the near future to support directed or valued networks. Thanks!

**Comment by martinamorris** [comment]
> This makes me think that portions of tergmLite and/or EpiModel may (possibly) not work with directed networks. @sgoodreau @smjenness do you know if anything in either package (or
> related packages) is plausibly specific to undirected networks?
> 
> To be clear, getting through InitErgmTerm is only one piece of the puzzle; I can test networkLites and networks for equivalent InitErgmTerm behavior, but that alone isn't a guarantee
> that everything else will handle things appropriately.

yup, that's ok.  if EpiModel can't use a specific term because it applies 
to directed nets, it won't be in our course materials or research.

it may be in someone else's approach, as one approach to network epi is to 
model the transmission network (which is directed) rather than the 
partnership network (which is not).  @sgoodreau, i'm wondering if the 
phylo net analyses will ever take this approach.

**Comment by sgoodreau** [comment]
I'm pretty sure that the phylogenetics work is very unlikely to ever be modeled using the machinery of directed networks within statnet or EpiModel.  Typically one would model the dynamic contact/act network.  Then on top of that one would model transmissions occurring using the type of additional code we always use.  Then on top of that one would model evolution.  Then on top of that one might model some sort of sampling process and then estimate a transmission network.  

Lumping all of those steps together into a single directed network process misses so many crucial elements and details that I don't see ever wanting to do it.  Even just skipping over contacts and modeling the transmission network directly as a directed network seems too much.  

In short: no need to worry about supporting directed networks here from my perspective.

**Comment by chad-klumb** [comment]
So other than `triangle` and `gwesp(fixed=TRUE)`, do we need anything else right this minute?

**Comment by martinamorris** [comment]
No -- but if we're going to leave the list of terms available for `tergm-lite` mostly unchanged, except for these two additions, then we need an issue to remind us to come back and expand the list later.

**Comment by chad-klumb** [comment]
Okay.  Expanding the `networkLite` class is something I hope to do after this tergm stuff and adding more terms goes hand in hand with that.

**Comment by smjenness** [comment]
This issue is closed with #30
